### PR TITLE
feat: P4ADEV-459-service-consumer

### PR DESCRIPTION
**Pull request by LarissaASLeite**
#### List of Changes
- Added service consumer to send notifications to IO
- Unit Test

#### How Has This Been Tested?

- Pre-Deploy Test
    - [x] Unit
    - [ ] Integration (Narrow)
- Post-Deploy Test
    - [x] Isolated Microservice
    - [ ] Broader Integration
    - [ ] Acceptance
    - [ ] Performance & Load

#### Types of changes

- [ ] PATCH - Bug fix (backwards compatible bug fixes)
- [x] MINOR - New feature (add functionality in a backwards compatible manner)
- [ ] MAJOR - Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -44,6 +56,21 @@ spring:
             sasl.mechanism: \${KAFKA_NOTIFICATION_SASL_MECHANISM:}
             security.protocol: \${KAFKA_NOTIFICATION_SECURITY_PROTOCOL:}
             max.request.size: \${KAFKA_CONFIG_MAX_REQUEST_SIZE:1000000}
+        binders:
+          notificationConsumer-in-0:
```
Comment (antonioT90) [inline src/main/resources/application.yml:60]:
```suggestion
        binders:
          notificationQueue-out-0:
            producer:
              configuration:
                client.id: io-notification
                connections.max.idle.ms: \${KAFKA_IO_NOTIFICATION_CONNECTION_MAX_IDLE_TIME:180000}
                retry.backoff.ms: \${KAFKA_IO_NOTIFICATION_RETRY_MS:\${KAFKA_RETRY_MS:10000}}
                linger.ms: \${KAFKA_IO_NOTIFICATION_LINGER_MS:\${KAFKA_LINGER_MS:2}}
                batch.size: \${KAFKA_IO_NOTIFICATION_BATCH_SIZE:\${KAFKA_BATCH_SIZE:16384}}
          notificationConsumer-in-0:
```

**Review by antonioT90 — CHANGES_REQUESTED** [review]

File: src/main/java/it/gov/pagopa/payhub/ionotification/service/notify/IONotificationServiceImpl.java [hunk 2]
```diff
@@ -21,4 +54,74 @@ public void sendMessage(NotificationQueueDTO notificationQueueDTO) {
         ioNotificationProducer.sendNotification(notificationQueueDTO);
     }
 
+    @Override
+    public void sendNotification(NotificationQueueDTO notificationQueueDTO) {
+        Optional<IOService> ioService = retrieveIOService(notificationQueueDTO);
+        if (ioService.isPresent()) {
```
[the remark this answers is not in the document]
Comment (LarissaASLeite) [reply src/main/java/it/gov/pagopa/payhub/ionotification/service/notify/IONotificationServiceImpl.java:60]: The "else" branches are handled inside the private methods. The intention here was to find a "clean" exit for the KO cases without throwing any exceptions, ensuring that the queue wouldn't retry the message. I have improved some of the logs inside the private methods.

**Review by antonioT90 — APPROVED** [review]

File: src/main/java/it/gov/pagopa/payhub/ionotification/constants/IONotificationConstants.java [hunk 3]
```diff
@@ -4,6 +4,10 @@ public class IONotificationConstants {
     public static final String SERVICE_STATUS_CREATED = "CREATED";
     public static final String SERVICE_STATUS_REQUESTED = "REQUESTED";
     public static final String SERVICE_STATUS_DELETED = "DELETED";
+    public static final String NOTIFICATION_STATUS_OK = "OK";
+    public static final String NOTIFICATION_STATUS_KO_SERVICE_NOT_FOUND = "KO_SERVICE_NOT_FOUND";
+    public static final String NOTIFICATION_STATUS_KO_SENDER_NOT_ALLOWED = "KO_SENDER_NOT_ALLOWED";
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/payhub/ionotification/constants/IONotificationConstants.java:9]: what about grouping those different constants representing multiple status into Enums to logically group together a set of status values for an entity?

File: src/main/java/it/gov/pagopa/payhub/ionotification/dto/mapper/IONotificationMapper.java [hunk 4]
```diff
@@ -17,4 +22,20 @@ public NotificationDTO mapToQueue(String fiscalCode, Long timeToLive, String sub
...
+        return FiscalCodeDTO.builder()
+                .fiscalCode(notificationQueueDTO.getFiscalCode())
+                .build();
+    }
+
+    public IONotification mapToSaveNotification(NotificationQueueDTO notificationQueueDTO, String status){
+        return IONotification.builder()
+                .notificationDate(LocalDateTime.now())
+                .userId(notificationQueueDTO.getFiscalCode())
+                .notificationStatus(status)
```
Comment (and-mora) [inline src/main/java/it/gov/pagopa/payhub/ionotification/dto/mapper/IONotificationMapper.java:36]: may it be appropriate to use the notification status enum in the models? if you think the refactor is too much extended, maybe you can move it to a new PR. But overall codify the status as a String is weak and not type safe

**Review by and-mora — APPROVED** [review]

**Pull request merged**